### PR TITLE
ci: configure connectivity test in delegated ipam e2e

### PR DESCRIPTION
**Pull request by wedaly**
Delegated IPAM E2E tests were using `cilium connectivity test` with no args. Pass additional args to capture sysdump on failure, run with higher test concurrency, and output junit file, and explicitly set external target/IP/CIDR.

### CODE REVIEW

File: .github/workflows/conformance-delegated-ipam.yaml [hunk 1]
```diff
@@ -335,7 +335,8 @@ jobs:
 
       - name: Cilium connectivity test
         run: |
-          cilium connectivity test
+          cilium connectivity test ${{ steps.vars.outputs.connectivity_test_defaults }} \
```
Comment (wedaly) [inline .github/workflows/conformance-delegated-ipam.yaml:338]: `connectivity_test_defaults` was being set above on line 163, but wasn't being used anywhere. My mistake!

**Comment by wedaly** [comment]
seems like this broke the upload junit step with:
> Error: Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run

investigating why...

**Comment by wedaly** [comment]
updated, now upload-artifacts is working
https://github.com/wedaly/cilium/actions/runs/12245376052

**Review by aanm — APPROVED** [review]

**Pull request merged**